### PR TITLE
HDDS-7082. Delete out of date audit logs

### PR DESCRIPTION
**Pull request by symious**
## What changes were proposed in this pull request?

This ticket is to add delete action to log4j to delete our of date audit logs.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7082

## How was this patch tested?

Test in cluster.

### CODE REVIEW

**Comment by kerneltime** [comment]
@aswinshakil

**Comment by kerneltime** [comment]
@symious will this be just 30 days independent of size?

**Comment by kerneltime** [comment]
@symious do you want to add a documentation page for audit config? It would be good to inform you that by default, we have the config set up for 30 days.
The only other concern I heard is about what the default value should be. If there are no other opinions, we can merge this in. cc @errose28

**Comment by symious** [comment]
@kerneltime Thanks for the suggestion. Are you suggesting adding a new md file in module of "hadoop-hdds-docs", which content package should I put the new audit log document? "Feature"?

**Comment by kerneltime** [comment]
I think it would be good to document audit as a feature somewhere in the official docs https://ci-hadoop.apache.org/view/Hadoop%20Ozone/job/ozone-doc-master/lastSuccessfulBuild/artifact/hadoop-hdds/docs/public/tools/auditparser.html is the mention of the audit. This can be a separate jira but would be good to explain the config and audit feature for Ozone.

**Comment by symious** [comment]
Checked the module of "hadoop-hdds-docs", seems the package of "tools" is for command line tools?
Maybe we can add a new page named "AuditLog in Ozone" like "Quota in Ozone" under "Feature" package?

**Comment by errose28** [comment]
Ozone audit logs and system logs use log4j for the configuration, so I think log4j config docs are the better reference for that rather than duplicating the info in our docs. If the default is changed to 30 days we could also just add a release note to 1.3.0, which might be more obvious than a new docs page. Something like the bullet entries [here](https://ozone.apache.org/release/1.2.1/)

File: hadoop-ozone/dist/src/shell/conf/dn-audit-log4j2.properties [hunk 1]
```diff
@@ -77,6 +77,14 @@ appender.rolling.policies.time.type=TimeBasedTriggeringPolicy
 appender.rolling.policies.time.interval=86400
 appender.rolling.policies.size.type=SizeBasedTriggeringPolicy
 appender.rolling.policies.size.size=64MB
+appender.rolling.strategy.type=DefaultRolloverStrategy
```
Comment (aswinshakil) [inline hadoop-ozone/dist/src/shell/conf/dn-audit-log4j2.properties:80]: Can you add comments on the change similar to the comments above? Same for other changes as well.

**Review by aswinshakil — COMMENTED** [review]
The change looks good to me. I just have one minor comment.

File: hadoop-ozone/dist/src/shell/conf/dn-audit-log4j2.properties (hunk 1 above)
Comment (symious) [reply hadoop-ozone/dist/src/shell/conf/dn-audit-log4j2.properties:82]: @aswinshakil Thanks for the review. Comments added.

**Comment by adoroszlai** [comment]
Do we need to set these configs as default, or is it enough to document how to configure it?

**Comment by swamirishi** [comment]
@kerneltime @errose28 @aswinshakil Can this be merged?

**Comment by errose28** [comment]
> Do we need to set these configs as default, or is it enough to document how to configure it?

I think @adoroszlai's suggestion is valid. A docs update with some examples of which log4j property files are relevant, some examples of common config changes like this one, and a link to the full log4j docs would probably be more helpful to an end user than the changes in this patch IMO.

**Comment by kerneltime** [comment]
> > Do we need to set these configs as default, or is it enough to document how to configure it?
> 
> I think @adoroszlai's suggestion is valid. A docs update with some examples of which log4j property files are relevant, some examples of common config changes like this one, and a link to the full log4j docs would probably be more helpful to an end user than the changes in this patch IMO.

@symious can you please address the outstanding comment and we can merge this in.

**Comment by symious** [comment]
Added a document named LogsInOzone.md, could you have a look?

**Pull request merged**